zebra/zebra_vty.c

```diff
@@ -1198,7 +1198,8 @@ static void vty_show_ip_route_detail(struct vty *vty, struct route_node *rn,
 				break;
 			}
 
-			if (re->vrf_id != nexthop->vrf_id) {
+			if ((re->vrf_id != nexthop->vrf_id)
+			     && !NEXTHOP_TYPE_BLACKHOLE) {
```

Comment:
@dslicenc shouldn't this be nexthop->type != NEXTHOP_TYPE_BLACKHOLE?

Reply:
Makes sense.  Will verify and update the PR.